cpu/atmega_common/periph/timer.c

```diff
@@ -136,9 +137,34 @@ int timer_set_absolute(tim_t tim, int channel, unsigned int value)
         return -1;
     }
 
+    unsigned state = irq_disable();
     ctx[tim].dev->OCR[channel] = (uint16_t)value;
     *ctx[tim].flag &= ~(1 << (channel + OCF1A));
     *ctx[tim].mask |= (1 << (channel + OCIE1A));
+    irq_restore(state);
+
+    return 0;
+}
+
+int timer_set(tim_t tim, int channel, unsigned int timeout)
+{
+    if (channel >= TIMER_CHANNELS) {
+        return -1;
+    }
+
+    unsigned state = irq_disable();
+    unsigned absolute = ctx[tim].dev->CNT + timeout;
+    ctx[tim].dev->OCR[channel] = absolute;
+    *ctx[tim].mask |= (1 << (channel + OCIE1A));
```

Comment:
```suggestion
    const unsigned mask = (1 << (channel + OCIE1A);
    *ctx[tim].mask |= mask;
```

Not sure if the compiler does this optimization on it's own, but it's more readable either way.

Reply:
Done

Reply:
Now that you pointed out my mistake I must say I liked it better before…
Because now there is this random mask variable when everywhere else that shift pattern is used…
And if we can't reuse the mask, there is nothing to gain.